### ISSUE TITLE
ancestor-comparison plot too brittle

### ISSUE THREAD

**Issue by jeromekelleher** [tskit-dev/tsinfer#83]
Two different problems here:
```
$ python3 evaluation.py  ac  -n 100  -s 12345
2018-07-17 14:20:13,075 [29332] CRITICAL root: Traceback (most recent call last):
  File "evaluation.py", line 1069, in <module>
    args.runner(args)
  File "evaluation.py", line 605, in run_ancestor_comparison
    assert np.array_equal(n_ton_table, test_freqs)
AssertionError
```

and
```
$ python3 evaluation.py  ac  -n 10 -d tmp__NOBACKUP__/new/ -X log -e 0.00 -s 12345 
2018-07-17 14:21:43,594 [29380] WARNING  py.warnings: /home/jk/.local/lib/python3.5/site-packages/matplotlib/colorbar.py:871: RuntimeWarning: invalid value encountered in true_divide
  z = np.take(y, i0) + (xn - np.take(b, i0)) * dy / db

2018-07-17 14:21:43,892 [29380] CRITICAL root: Traceback (most recent call last):
  File "evaluation.py", line 1069, in <module>
    args.runner(args)
  File "evaluation.py", line 659, in run_ancestor_comparison
    save_figure(name_format.format("length-scatter_{}".format(colorscale.lower())))
  File "evaluation.py", line 35, in save_figure
    plt.savefig(basename + "." + _output_format)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/pyplot.py", line 697, in savefig
    res = fig.savefig(*args, **kwargs)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/figure.py", line 1572, in savefig
    self.canvas.print_figure(*args, **kwargs)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/backend_bases.py", line 2244, in print_figure
    **kwargs)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/backends/backend_agg.py", line 545, in print_png
    FigureCanvasAgg.draw(self)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/backends/backend_agg.py", line 464, in draw
    self.figure.draw(self.renderer)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/figure.py", line 1143, in draw
    renderer, self, dsu, self.suppressComposite)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/image.py", line 139, in _draw_list_compositing_images
    a.draw(renderer)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/axes/_base.py", line 2409, in draw
    mimage._draw_list_compositing_images(renderer, self, dsu)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/image.py", line 139, in _draw_list_compositing_images
    a.draw(renderer)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/axis.py", line 1150, in draw
    self.label.draw(renderer)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/artist.py", line 63, in draw_wrapper
    draw(artist, renderer, *args, **kwargs)
  File "/home/jk/.local/lib/python3.5/site-packages/matplotlib/text.py", line 762, in draw
    raise ValueError("posx and posy should be finite values")
ValueError: posx and posy should be finite values
```

Can you take a look at these please @hyanwong and try the script for a variety of parameter values?

**Comment by hyanwong** [comment]
Will do

**Comment by hyanwong** [comment]
The first one is interesting, and I think properly catches an error we made. We assumed that `anc.ancestors_time[:] + 1` for an inferred sequence gives the frequency of the ancestral focal site, but actually, the time simply increments from 1..x. So if there is a "missing" frequency: e.g. no alleles at a  frequency of 42, then the (ancestors_time +1) value will jump straight from 41 to 43. 

I.e. the hack that we used to get frequency from the inferred times is fragile, and we probably shouldn't use it.
